clusterctl/clusterdeployer/clusterapiserver.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterdeployer
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"text/template"
+
+	"github.com/golang/glog"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/cert/triple"
+)
+
+var apiServerImage = "gcr.io/k8s-cluster-api/cluster-apiserver:0.0.3"
+
+func init() {
+	if img, ok := os.LookupEnv("CLUSTER_API_SERVER_IMAGE"); ok {
```

Comment:
Why an env var instead of a flag?

Reply:
It is what we have been doing so far in this repo. We can change it to a flag post v0 if folks really prefer a flag over a environment variable for a development hook.